pkg/ebpf/network_filter.go

```diff
@@ -0,0 +1,209 @@
+package ebpf
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// ConnectionFilter holds a user-defined blacklisted IP/CIDR, and ports
+type ConnectionFilter struct {
+	IP       *net.IPNet
+	Ports    map[uint16]ConnTypeFilter
+	AllPorts ConnTypeFilter
+}
+
+// ConnTypeFilter holds user-defined protocols
+type ConnTypeFilter struct {
+	TCP bool
+	UDP bool
+}
+
+// ParseConnectionFilters takes the user defined blacklist and returns a slice of ConnectionFilters
+func ParseConnectionFilters(filters map[string][]string) (blacklist []*ConnectionFilter) {
+	for ip, ports := range filters {
+		filter := &ConnectionFilter{Ports: map[uint16]ConnTypeFilter{}}
+		var subnet *net.IPNet
+		var err error
+
+		// retrieve valid IPs
+		if strings.ContainsRune(ip, '*') {
+			subnet = nil // use for wildcard
+		} else if strings.ContainsRune(ip, '/') {
+			_, subnet, err = net.ParseCIDR(ip)
+		} else if strings.ContainsRune(ip, '.') {
+			_, subnet, err = net.ParseCIDR(ip + "/32") // if given ipv4, prefix length of 32
+		} else if strings.Contains(ip, "::") {
+			_, subnet, err = net.ParseCIDR(ip + "/64") // if given ipv6, prefix length of 64
+		} else {
+			log.Debugf("Invalid IP/CIDR/* defined for connection filter: %s", err)
+			continue
+		}
+
+		if err != nil {
+			log.Errorf("Given filter will not be respected. Could not parse given IPs: %s", err)
+			continue
+		}
+		filter.IP = subnet
+
+		validFilter := true
+		for _, v := range ports {
+			lowerPort, upperPort, connTypeFilter, valid := parsePortAndProtocolFilter(v)
+			validFilter = valid
+
+			if !validFilter {
+				break
+			}
+
+			// port is wildcard
+			if lowerPort == 0 && upperPort == 0 && validFilter {
+				if subnet == nil && connTypeFilter.TCP && connTypeFilter.UDP {
+					log.Errorf("Given rule will not be respected. Invalid filter with IP/CIDR as * and port as *: %s", err)
+					validFilter = false
+					break
+				}
+				filter.AllPorts.TCP = connTypeFilter.TCP || filter.AllPorts.TCP
+				filter.AllPorts.UDP = connTypeFilter.UDP || filter.AllPorts.UDP
+
+				continue
```

Comment:
nit: can we use 

if lowerPort == 0 && upperPort == 0 && validFilter {
  ... 
} else if upperPort == 0 {

}. else {

}
```

instead of using `if` and `continue`